src/test/java/com/epam/poc/homePage/SearchFunctionalityTest.java

```diff
@@ -0,0 +1,100 @@
+package com.epam.poc.homePage;
+
+import com.epam.poc.commons.BaseTest;
+import com.epam.poc.pageObjects.HeaderPageObject;
+import com.epam.poc.pageObjects.SearchResultObject;
+import com.epam.poc.pageObjects.homepage.HomePageObject;
+import com.epam.poc.utilities.listeners.TestListener;
+import io.qameta.allure.*;
+import org.openqa.selenium.By;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import static com.epam.poc.pageUIs.HeaderPageUI.*;
+import static com.epam.poc.pageUIs.SearchResultUI.*;
+
+@Listeners({TestListener.class})
+@Epic("Regression test")
+@Feature("SearchFunctionality")
+public class SearchFunctionalityTest extends BaseTest {
+
+    private HeaderPageObject headerPage;
+    private SearchResultObject searchResultPage;
+
+    @BeforeMethod
+    public void beforeMethod() {
+        homePage = new HomePageObject(driver);
+        headerPage = new HeaderPageObject(driver);
+        searchResultPage = new SearchResultObject(driver);
+
+        homePage.closeHomePagePopup();
+    }
+
+    @Test
+    @Description("Test description: Verify search with keyword 'Túi xách'")
+    @Story("Search Functionality")
+    public void verifySearchKeywordTuiXach() {
+        headerPage.sendKeyToElement(driver, By.cssSelector(SEARCH_TEXTBOX_CSS), "túi xách");
+        headerPage.clickToElement(driver, By.cssSelector(SEARCH_BUTTON_CSS));
+
+        String shopRelatedLabelExpected = "Shop liên quan đến \"túi xách\"";
+        Assert.assertEquals(searchResultPage.getElementText(driver,
+                By.cssSelector(SHOP_RELATED_LABEL_CSS)), shopRelatedLabelExpected);
+
+        String searchResultLabelExpected = "Kết quả tìm kiếm cho từ khoá 'túi xách'";
+        Assert.assertEquals(searchResultPage.getElementText(driver,
+                By.cssSelector(SEARCH_RESULT_LABEL_CSS)), searchResultLabelExpected);
```

Comment:
the PageUI should move to PageObject.
In test class, PageUI should not be present